pkgs/development/web/bun/default.nix

```diff
@@ -64,4 +64,20 @@ stdenvNoCC.mkDerivation rec {
     maintainers = with maintainers; [ DAlperin jk ];
     platforms = builtins.attrNames dists;
   };
+
+  updateScript = writeShellScript "update-bun" ''
```

Comment:
You also need to take `writeShellScript` and the other packages as an argument to the package function, otherwise they will not be defined and you will not be even able to do `nix-build -A bun`.

By the way, to test, you can run `nix-shell maintainers/scripts/update.nix --argstr path bun`